core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/RecordingTaskManagerListener.java

```diff
@@ -49,14 +49,14 @@ public RecordingTaskManagerListener(DiscoveryNode localNode, String... actionMas
     @Override
     public synchronized void onTaskRegistered(Task task) {
         if (Regex.simpleMatch(actionMasks, task.getAction())) {
-            events.add(new Tuple<>(true, task.taskInfo(localNode, true)));
+            events.add(new Tuple<>(true, task.taskInfo(localNode, System.currentTimeMillis(), true)));
```

Comment:
So I believe System.nanoTime is supposed to be better for measuring the time that a task has been running. It never goes backwards, iirc. And currentTimeMillis is totally allowed to do that. They do that by making it an arbitrary, can't convert me to a wall clock time number. So maybe we should use that for the run time? That is yet another reason we'd want to throw it over the wire - you can't sync that time with any sort of time sync. That is kind of its point.